File: 251001/Korotkaya/Publisher/DTO/Editor.cs
﻿namespace WebApplication1.DTO
{
    public class EditorRequestTo
    {
        public long? Id { get; set; }
        public string Login { get; set; } = string.Empty;
        public string Password { get; set; } = string.Empty;
        public string Firstname { get; set; } = string.Empty;
        public string Lastname { get; set; } = string.Empty;
    }

    public class EditorResponseTo
    {
        public long id { get; set; }
        public string login { get; set; }
        public string firstname { get; set; }
        public string lastname { get; set; }
    }
}

File: 251003/Belevich/Publisher/src/main/java/com/bsuir/dc/model/Editor.java
package com.bsuir.dc.model;

import jakarta.persistence.*;

@Entity
@Table(name = "tbl_editor")
public class Editor {
    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(name = "login", unique = true)
    private String login;

    @Column(name = "password")
    private String password;

    @Column(name = "firstname")
    private String firstname;

    @Column(name = "lastname")
    private String lastname;

    public void setId(long id) { this.id = id; }
    public long getId() { return id; }

    public void setLogin(String login) { this.login = login; }
    public String getLogin() { return login; }

    public void setPassword(String password) { this.password = password; }
    public String getPassword() { return password; }

    public void setFirstname(String firstname) { this.firstname = firstname; }
    public String getFirstname() { return firstname; }

    public void setLastname(String lastname) { this.lastname = lastname; }
    public String getLastname() { return lastname; }
}
